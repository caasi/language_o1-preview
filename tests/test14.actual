Result is not a number.
Statement List:
  ADT Definition: Maybe
    ADT Constructor Definition: Just
      Field 1:
        Number (Basic)
    ADT Constructor Definition: Nothing
  Let Binding: value
    Value:
      ADT Constructor Call: Just
        Argument 1:
          Number: 10.000000
    Body:
      Variable: value
