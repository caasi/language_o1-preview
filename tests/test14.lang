type Maybe = Just Number | Nothing

let value = Just 10 in value

