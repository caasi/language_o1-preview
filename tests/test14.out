Just (10.000000)
